### ISSUE TITLE
DIGITS ResNet-50 Training on Cifar-10

### ISSUE THREAD

**Issue by adeely9** [NVIDIA/DIGITS#1860]
Getting error on training ResNet-50 Kaiming He model on Cifar-10 Dataset

Error: ERROR: >= 0 (-5 vs. 0) on pool5 layer.
Creating layer res5c
Creating Layer res5c
res5c <- res5b_res5b_relu_0_split_1
res5c <- res5c_branch2c
res5c -> res5c
Setting up res5c
Top shape: 12 2048 1 1 (24576)
Memory required for data: 56672304
Creating layer res5c_relu
Creating Layer res5c_relu
res5c_relu <- res5c
res5c_relu -> res5c (in-place)
Setting up res5c_relu
Top shape: 12 2048 1 1 (24576)
Memory required for data: 56770608
Creating layer pool5
Creating Layer pool5
pool5 <- res5c
pool5 -> pool5
>= 0 (-5 vs. 0)

Is this my GPU compatibility issue or something else. 

My caffe_output.log file is also attached.
[caffe_output (1).log](https://github.com/NVIDIA/DIGITS/files/1404405/caffe_output.1.log)

I'm new to DIGITS and caffe, totally exhausted now! 
Your quick help will be appreciated Sir @lukeyeager

**Comment by ethantang95** [comment]
To be honest, I am not even sure if ResNet can train well with CiFar-10 data set because it is way smaller than the input (ResNet-50 takes in 224x224 I believe with respect to the original paper). I have tried training CiFar10 with ResNet-34 on Tensorflow and the results that it gives are very bad.

The original paper uses a different network to train CiFar even with different feature map sizes per residual set.

You can feel free to explore my ResNet-34 tensorflow implementation if you are curious, although I am not sure if the bad performance is due to it not being optimized for CiFar10, or there are problems with my implementation.

My ResNet-34 implementation is at: https://github.com/ethantang95/DIGITS/blob/personal-work/extra-tf-networks/resnet34.py

**Comment by adeely9** [comment]
Thanks for your reply! 
I was also thinking about that, the thing I explored till now is, there is some kernel_size issue in pool5 layer.

My basic purpose of training renet50 on cifar-10 is to learn fine-tuning, training and testing procedures as a starter.
Will you please suggest me any other Model which could be trained, tested and fine-tuned on Cifar-10 Dataset?

**Comment by ethantang95** [comment]
Alex net and Googlenet are good ones to start with

**Comment by adeely9** [comment]
I just tried GoogleNet Caffe model for CIFAR-10 Dataset and changed input_parameters
from: input_param { shape: { dim: 10 dim: 3 dim: 224 dim: 224 } }
to: input_param { shape: { dim: 10 dim: 3 dim: 32 dim: 32 } } 

and classifier num_output from 1000 to 10.

And I am again getting the same error: >= 0 (-5 vs. 0) on pool5 layer.

Sir @lukeyeager  please help me out what I am doing wrong in process?
Anyone who could help me?
